### ISSUE TITLE
Python 2.7.15 Set Settings

### ISSUE THREAD

**Issue by blacktwin** [pkkid/python-plexapi#314]
In my attempts to update the OnDeckWindow with python 2.7:

```python
plex = PlexServer(PLEX_URL, PLEX_TOKEN)
print(plex.settings.OnDeckWindow)
plex.settings.get('OnDeckWindow').set(26)
plex.settings.save()
plex = PlexServer(PLEX_URL, PLEX_TOKEN)
print(plex.settings.OnDeckWindow)
```
I would receive a `basestring` type error:
```
    plex.settings.get('OnDeckWindow').set(26)
  File "C:\Python27\lib\site-packages\plexapi\settings.py", line 155, in set
    self._setValue = tostr(value)
TypeError: The basestring type cannot be instantiated
```

From my minimal researching basestring type was added in python 3. Running the above script with python 3 works fine. Finding the basestring call in compat.py. Modifying compat.py to check for python versions less than 3 seems to resolve the issue.
Before:
```python
try:
    string_type = basestring
except NameError:
    string_type = str
```

Update:
```python
try:
    if version_info < (3,):
        string_type = unicode
    else:
        string_type = basestring
except NameError:
    string_type = str
```
Not sure if this will break anything and wasn't sure if a PR would be appropriate.

**Comment by andrey-yantsen** [comment]
We shouldn't apply this patch, actually. And tests wouldn't fail, probably.

`basestring` class doesn't expected to be instantiated, it's just a base class for `str` and `unicode`, and that's exactly what the exception says :)

Currently there are 3 usages of string_type: 
![image](https://user-images.githubusercontent.com/629281/48660661-3103f080-ea5d-11e8-9482-807265c9360d.png)

1. Converting to str in settings.py
2. Testing for `is string` in conftest.py
3. Testing for `is string` in SecretsFilter in utils.py

And suggested fix will fix only first usage, but in the latter 2 we may have some unexpected behaviour.

**Comment by Hellowlol** [comment]
Yeah I agree. I havnt touched this part of the code but I think there are some bugs in the types. Feel free to send a pr or I’ll get to this later.

Btw. Do you know why the tests are failing for the smart branch? I understand there is a issue with the docs string but not what I need to do to get it fixed.

**Comment by andrey-yantsen** [comment]
@Hellowlol I've took a look on your branch and found the error :) you've missed terminating apostrophe in `section` and `returns`

**Comment by Hellowlol** [comment]
Thanks!

**Comment by dannycvera** [comment]
I'm currently getting the same issue with customConnections.
Trying to use ngrok to get around Carrier Grade NAT, but need to update the Custom URL in a script.

when running :
plex.settings.get('customConnections').set('http://s0m3t3xt.ngrok.io:80')

I receive the error:

  File "/usr/lib/python2.7/site-packages/plexapi/settings.py", line 152, in set
    self._setValue = tostr(value)
TypeError: The basestring type cannot be instantiated

tried adding .toUrl() at the end with same error.

**Comment by senorgeneral** [comment]
I'm currently getting the same issue with customConnections.
Trying to use ngrok to get around Carrier Grade NAT, but need to update the Custom URL in a script.

when running :
plex.settings.get('customConnections').set('http://s0m3t3xt.ngrok.io:80')

I receive the error:

  File "/usr/lib/python2.7/site-packages/plexapi/settings.py", line 152, in set
    self._setValue = tostr(value)
TypeError: The basestring type cannot be instantiated

tried adding .toUrl() at the end with same error.

**Comment by andrey-yantsen** [comment]
I assume it was fixed in e5df09ca6cb5f4bdf311766de5fd39606ee33070, @blacktwin & @senorgeneral can you please confirm? Unfortunately no releases were created since November, so you need to check it in master branch.

**Comment by blacktwin** [comment]
@andrey-yantsen 

`C:\Python27>pip install git+git://github.com/pkkid/python-plexapi.git@e5df09ca6cb5f4bdf311766de5fd39606ee33070 --force-reinstall`

And reran the code.

```python
print(plex.settings.OnDeckWindow)
plex.settings.get('OnDeckWindow').set(26)
plex.settings.save()
plex = PlexServer(PLEX_URL, PLEX_TOKEN)
print(plex.settings.OnDeckWindow)
```
output:
```
26
26
```

`pip install plexapi --force-reinstall`

output:
```
    plex.settings.get('OnDeckWindow').set(26)
  File "C:\Python27\lib\site-packages\plexapi\settings.py", line 152, in set
    self._setValue = tostr(value)
TypeError: The basestring type cannot be instantiated
```

**Comment by senorgeneral** [comment]
Changing settings only works with Python 3.
Have to install plexapi using pip3 and it will work. Can't say whether anything else is broken when using the latest python.

**Comment by pkkid** [comment]
@andrey-yantsen - Do we need another release to pypi to fix this?

**Comment by andrey-yantsen** [comment]
@pkkid yeah, would be great

**Comment by blacktwin** [comment]
I think this can be closed as 2.7 is no longer supported. Feel free to close.
